feat: support panning and zomming on image view; use pixi

sehilyi opened the pull request.

sehilyi commented on server/flask_server/api.py:
```diff
@@ -278,7 +278,8 @@ def get_matrix_sample(**kwargs):
 
-def get_IDC_sample(id):
+def get_IDC_sample(**kwargs):
+    id = kwargs['id']
```
To avoid a Python type error, I had to change this

wangqianwen0418 replied:
Ha, my mistake. Must accidentally left this error

sehilyi commented on front/src/components/ImageContext.tsx:
```diff
@@ -36,65 +44,112 @@ const ImageContext = (props: Props) => {
-        ctx.globalAlpha = 1.0;
-      });
-  }, [samples]);
+  // useEffect(() => {
+  //   if (!canvasRef.current) return;
+  //   const ctx = canvasRef.current.getContext('2d')!;
+  //   imgSamples
+  //     .filter(d => d.filtered)
+  //     .forEach(sample => {
+  //       ctx.beginPath();
+  //       ctx.globalAlpha = 0.1;
+  //       ctx.fillStyle = 'white';
+  //       ctx.fillRect(sample.x - offsetX, sample.y - offsetY, imgSize, imgSize);
+  //       ctx.globalAlpha = 1.0;
+  //     });
+  // }, [samples]);
```
I still need to implement this. Would prefer to work on this in a separate PR.

sehilyi commented on front/src/App.css:
```diff
@@ -36,6 +36,10 @@
 }
 
+#imageContainer .ant-card-body {
+  padding: 0px !important;
```
Need to add this to remove a padding in Image Context Card

wangqianwen0418 commented:
@sehilyi, thanks so much for this PR! It looks great.

> I still need to implement this. Would prefer to work on this in a separate PR if that's okay.

Sure, no problem! I will merge the two PRs together later as I still need the filter function in the main branch

sehilyi commented:
@wangqianwen0418, I added the filtering functionality (refer to the [last commit](https://github.com/wangqianwen0418/DRAVA/pull/14/commits/0ce5609f764ea8183a15f99401420cb91643c189)). I used the same approach that was supported previously, i.e., overlaying semi-transparent rectangles. I also fixed a bug that filtering overlays were not correctly removed when a user deselects a bar on the histogram.

![Screen Shot 2022-08-06 at 10 54 19](https://user-images.githubusercontent.com/9922882/183254415-6ab73272-2130-4cd3-972c-be135bb3bccd.png)

![Screen Shot 2022-08-06 at 10 54 27](https://user-images.githubusercontent.com/9922882/183254414-d1ebc70d-5579-4bcd-a6f1-9dc5db4508f3.png)

wangqianwen0418 commented on front/src/components/ImageContext.tsx:
```diff
@@ -78,36 +79,39 @@ const ImageContext = (props: Props) => {
-      // pixiRoot.current?.removeChildren();
+      pixiRoot.current?.removeChildren();
+      pixiFilter.current.removeChildren();
       // pixiRoot.current?.destroy();
```
Can we remove these commented code line 108-111?

wangqianwen0418 commented:
Thanks @sehilyi for this PR! Tested locally and it works well!
